lib/graphql/schema/validator/required_validator.rb

```diff
@@ -51,27 +51,34 @@ def initialize(one_of: nil, argument: nil, message: nil, **default_options)
         end
 
         def validate(_object, context, value)
-          matched_conditions = 0
+          fully_matched_conditions = 0
+          partially_matched_conditions = 0
 
           if !value.nil?
             @one_of.each do |one_of_condition|
               case one_of_condition
               when Symbol
                 if value.key?(one_of_condition)
-                  matched_conditions += 1
+                  fully_matched_conditions += 1
                 end
               when Array
-                if one_of_condition.all? { |k| value.key?(k) }
-                  matched_conditions += 1
-                  break
```

Comment:
This `break` statement is what caused the order dependency issue. I looked in the Git history to see why it was present. It existed in the original commit, without an explicit explanation. Upon further evaluation and experimentation, I discovered it led to the order dependency mentioned. Removing it outright fixes this order depedency.

Reply:
Funny, we have an internal helper called `validates_at_least_one_of` that was exploiting your original thinking, Robert. :)